py-polars/src/conversion/any_value.rs

```diff
@@ -419,3 +420,12 @@ pub(crate) fn py_object_to_any_value(ob: &PyAny, strict: bool) -> PyResult<AnyVa
         })
     })
 }
+
+fn any_values_to_supertype_and_n_types(values: &[AnyValue]) -> PolarsResult<(DataType, usize)> {
```

Comment:
Can you make this generic? Then `fn any_values_to_supertype` can dispatch to this one and just get the `DataType` out.

Reply:
I solved it slightly differently (extract the logic and re-use).